### ISSUE TITLE
Feature Request - Add Pressure Trend

### ISSUE THREAD

**Issue by chasut** [briis/hass-weatherflow2mqtt#18]
Only sensor I am missing from the SmartWeather integration.  TIA for considering the feature.

**Comment by GlennGoddard** [comment]
Have you considered a trend or derivative sensor for HA?
I used the derivative sensor for pressure and graphed it out to see the trend over time.
The derivative will show you the rate of change which is slightly more useful than trend in my opinion.
This being in my sensor.yaml which is an include in configuration.yaml
`- platform: derivative
  source: sensor.purpleair_pressure_2
  name: Barometric Pressure Change
  unit_time: h
  time_window: "06:00:00"
  round: 2
`
For the trend sensor, you would be under binary_sensor or things like the custom-bar-card will do the trend for you (not always pretty thou)

**Comment by briis** [comment]
Glenn is right that this can be done in HA, by creating a sensor, but if it has to be done in this program, I will need to store the pressure values for at least the last three hours, as the way WeatherFlow has defined this is *The Pressure Trend description is determined by the rate of change over the past 3 hours.*

To accomplish this, and maybe future trends and sensors (High Sensor Value for the day, Low Sensor value for the day, etc.) I think I will convert the simple storage system I have today, to a SQLite Database, which makes this easier to implement and faster to to retrieve the right results.

Thoughts?

**Comment by chasut** [comment]
Your call on including in the calculated sensor.  I am totally fine either way.  It does get into a bit of a grey area if you need to add additional infrastructure to what is a relatively simple and clean translator/repeater.  There is a good argument to keep the number of moving parts small so that it is as reliable as possible.  After all, one of the main reasons to go with this approach over the web version is reliability in the case that the internet or the rest endpoints are down.  So from that perspective I would say just leave that up to an HA template sensor.

Not sure what you have planned for the dBase, but If you do decide to include it for completeness, I would maybe suggest just keeping the running total and times in the local storage vs. all of the individual observation history.  If the program dies or you lose readings for an extended period, you will have to start the sum all over again anyway.  If it is a short outage, or just a few dropped readings, you are no worse off without the detail.  If it is just the i/o that is the concern, then back to just keeping it simple.

Anyway, just my 2 cents.

Glenn, thanks for the tip on the derivative platform.  I didn't know that existed till you pointed it out.  Very handy, I have been doing things the hard way!  Always learning something new.

**Comment by GlennGoddard** [comment]
I'm torn...I see the benefits & drawbacks of both.  This will be much more simple than the current condition logic.
I posted in the HA forum about my attempt to mimic the WeatherFlow App inside Home Assistant.

**Comment by chasut** [comment]
That was too easy.  Took Glenn's tip and created my first derivative sensor:

```
  - platform: derivative
    source: sensor.sea_level_pressure
    name: pressure_trend_value
    round: 3
    unit_time: h 
    time_window: "03:00:00"  
```
Then just a simple template sensor to translate into the WeatherFlow output format...
```
      current_weather_pressure_trend:
        friendly_name: Pressure Trend
        value_template:  "{% if states('sensor.pressure_trend_value')|float > 0.0295 %} rising
                          {%- elif states('sensor.pressure_trend_value')|float < -0.0295 -%} falling
                          {%- else -%} steady
                          {% endif %}"
```

**Comment by briis** [comment]
> That was too easy. Took Glenn's tip and created my first derivative sensor:

There you go. 😁 I have spend the day re-writing the storage modules, so it will now use a SQLite database, instead of two text files. I did this for more than this reason, and SQLite is simple to use, and even simple to backup.

I have not changed the way the program works, I basically just re-wrote the *read* and *write* modules to use SQLite instead of flat text files. 

And while I was doing this, I have also created the Pressure Trend functions, so as soon as I have time, I will add this sensor - even though you don't need it anymore - but now it is done.

I will let you know when this is ready to test.

**Comment by GlennGoddard** [comment]
You say you have every other sensor....
Are you calculating these are am I missing where they are?  (yes, I realize the app was just updated to display some of these easier)
Where are you getting Wet Bulb temperature?
Delta-T (Which I'm not sure what difference it is measuring...Air Temp is 60F, DewPoint is 58F, Wet Bulb is 59F but Delta-T is 33F.
I noticed that Air Density is not tracking between App and the calculated.  App = 0.07578 lbs/ft^3 and HA = 0.09 lb/ft^3.  It is not limited to a rounding issue.

**Comment by GlennGoddard** [comment]
Just stumbled on this:
[https://weatherflow.github.io/Tempest/api/derived-metric-formulas.html](url)
Something is not right in the backend for WeatherFlows calculations on Delta-T

**Comment by briis** [comment]
> Are you calculating these are am I missing where they are? (yes, I realize the app was just updated to display some of these easier)

I am not sure I said I have all the sensors, but eventually, I can make some more calculated Sensors, based on the formulas that WeatherFlow describes in the section you mention. I take the formulas from that link and try and convert them to Python.

If Air density is off for you, then there must be something with my conversion to imperial units, as it is pretty much spot on for me. But I will look at the formula again.

**Comment by briis** [comment]
I have now released V2.0, where this is included. See the CHANGELOG.md for more info.
